physics/body_surface_frame_field_body.hpp

```diff
@@ -0,0 +1,55 @@
+﻿
+#pragma once
+
+#include "physics/body_surface_frame_field.hpp"
+
+#include "physics/continuous_trajectory.hpp"
+#include "quantities/quantities.hpp"
+
+namespace principia {
+namespace physics {
+namespace internal_body_surface_frame_field {
+
+using geometry::Bivector;
+using geometry::Displacement;
+using geometry::InnerProduct;
+using geometry::Normalize;
+using geometry::Wedge;
+using quantities::Length;
+
+template<typename Frame, typename ThisFrame>
+BodySurfaceFrameField<Frame, ThisFrame>::BodySurfaceFrameField(
+    Ephemeris<Frame> const& ephemeris,
+    Instant const& t,
+    not_null<RotatingBody<Frame> const*> const body)
+    : body_axis_(body->polar_axis()),
+      body_position_(
+          ephemeris.trajectory(body)->EvaluatePosition(t, /*hint=*/nullptr)) {}
+
+template<typename Frame, typename ThisFrame>
+Rotation<ThisFrame, Frame>
+BodySurfaceFrameField<Frame, ThisFrame>::FromThisFrame(
+    Position<Frame> const& q) const {
+  // A vector going from the centre of the body to the point in the field.
+  Displacement<Frame> const displacement = q - body_position_;
```

Comment:
A comment that reminds the reader of the definition of position subtraction isn't all that useful; call the variable `from_body_centre` or something like that, so that we have some idea what it is below.